paramak/parametric_components/hexagon_pin.py

```diff
@@ -0,0 +1,96 @@
+
+import math
+from typing import Optional, Tuple
+
+from paramak import ExtrudeStraightShape
+
+
+class HexagonPin(ExtrudeStraightShape):
+    """Creates an extruded hexagon by a provided distance about a center point.
+
+    Args:
+        length_of_side: the vertical (z axis) height of the coil (cm).
+        width: the horizontal (x axis) width of the coil (cm).
+        center_point: the center of the coil (x,z) values (cm).
+        stp_filename: defaults to "PoloidalFieldCoil.stp".
+        stl_filename: defaults to "PoloidalFieldCoil.stl".
+        name: defaults to "pf_coil".
+        material_tag: defaults to "pf_coil_mat".
+    """
+
+    def __init__(
+        self,
+        length_of_side: float,
+        distance: float,
+        center_point: Tuple[float, float] = (0, 0),
+        stp_filename: Optional[str] = "HexagonPin.stp",
+        stl_filename: Optional[str] = "HexagonPin.stl",
+        name: Optional[str] = "hexagon_pin",
+        material_tag: Optional[str] = "hexagon_pin_mat",
+        **kwargs
+    ) -> None:
+
+        super().__init__(
+            name=name,
+            distance=distance,
+            material_tag=material_tag,
+            stp_filename=stp_filename,
+            stl_filename=stl_filename,
+            **kwargs
+        )
+
+        self.center_point = center_point
+        self.length_of_side = length_of_side
+
+    @property
+    def center_point(self):
+        return self._center_point
```

Comment:
This line isn't tested.

Reply:
Lots more tests have now been added. There center point setting and getting is tested in these new tests. Ideally mores tests would be added but I think this component is above average now